### PR TITLE
EVG-5982 mute / unmute support.

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

File: public/static/app/common/MuteDataService.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+mciModule.factory('MuteDataService', function(
+  $log, Stitch, STITCH_CONFIG
+) {
+  // Get the mutes as a promise
+  // :param query: The query to evaluate.
+  // :param projection: Specify the fields in the returned documents.
+  const queryQ = (query, projection) => {
+    return Stitch.use(STITCH_CONFIG.PERF).query(function (db) {
+      // Remove Undefined values, null and false are acceptable.
```
Comment (dbradf) [inline public/static/app/common/MuteDataService.js]: This comment is a little confusing. It is difficult to tell what is being removed and what is acceptable. Maybe break it into two sentences?

(same for this comment in other places in this file)

File: public/static/app/perf-bb/PerfBBOutliersCtrl.js [hunk 2]
```diff
@@ -395,4 +190,373 @@ mciModule.controller('PerfBBOutliersCtrl', function (
...
+    }
+
+  }
+
+  return MuteHandler;
+}).factory('OutlierState', function(FORMAT, MDBQueryAdaptor, EvgUtil, EvgUiGridUtil, uiGridConstants, OutliersDataService, MuteDataService, $q) {
+  class OutlierState {
+    // Create a new state instance to encapsulate filtering, sorting and generating queries.
+    //
+    // :param str project: The project name, e.g. 'sys-perf'.
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBOutliersCtrl.js]: This is probably outside of the scope of this PR (and should probably involve the evergreen team), but we might want to consider standardizing on something like [JSDoc formatting](https://en.wikipedia.org/wiki/JSDoc) for javascript code comments.

File: public/static/app/perf-bb/PerfBBOutliersCtrl.js [hunk 3]
```diff
@@ -395,4 +190,373 @@ mciModule.controller('PerfBBOutliersCtrl', function (
...
+    }
+
+    // Handle a rowsRendered event. This is only expected to be called once.
+    // :param object api: The grid api reference.
+    // :param object cols: The column definitions.
+    onRowsRendered(api) {
+      this.api = api;
+      this.getCol = EvgUiGridUtil.getColAccessor(api);
+
+      //const setInitialGridFiltering = () => {
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBOutliersCtrl.js]: [nit] Can the commented out lines in this function be deleted?

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js [hunk 4]
```diff
@@ -2,21 +2,55 @@ describe('PerfBBOutliersCtrlTest', () => {
...
   const project = 'test-project';
 
   beforeEach(inject(function ($rootScope, $controller, $injector) {
-    // $scope, $window, EvgUiGridUtil, EvgUtil, FORMAT, MDBQueryAdaptor, uiGridConstants,
-    //     STITCH_CONFIG, Stitch,
     scope = $rootScope;
     format = $injector.get('FORMAT');
+    OutliersDataService = $injector.get('OutliersDataService');
+    $log = $injector.get('$log');
+    // $timeout = $injector.get('$timeout');
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: [nit] Can this commented out line be removed?

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js [hunk 5]
```diff
@@ -39,211 +73,1443 @@ describe('PerfBBOutliersCtrlTest', () => {
...
+
+      expect(_.debounce).toHaveBeenCalledWith(vm.reload, 500);
+      expect(_.bind).toHaveBeenCalledWith(vm.state.onRowsRendered, vm.state);
+      expect(_.once).toHaveBeenCalledWith(bind, api);
+    });
+
+  });
+
+});
+describe('PerfBBOutliersFactoriesTest', () => {
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: [nit] Can there be a blank line before this 'describe'?

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js [hunk 6]
```diff
@@ -39,211 +73,1443 @@ describe('PerfBBOutliersCtrlTest', () => {
...
+
+  beforeEach(module('MCI'));
+
+  const project = 'test-project';
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: [nit] 'project' and 'today' both look like top level constants. Can they be put in ALL_CAPS to be consistent with the other constants in this file?

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js [hunk 7]
```diff
@@ -39,211 +73,1443 @@ describe('PerfBBOutliersCtrlTest', () => {
...
+        spyOnProperty(state, 'sorting', 'get').and.returnValue('get sorting');
+
+        expect(state.getAggChain()).toEqual(['compileFiltering', 'compileSorting', {$limit: 10}]);
+      });
+
+    });
+
+    describe('sortRevision', () => {
+      let a, b, rowA, rowB;
+      // beforeEach(() => {
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: [nit] removed commented out code.

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js [hunk 8]
```diff
@@ -1,34 +1,24 @@
...
-        .ok('Ok')
-        .cancel('Cancel')
-        .title('Confirm')
-        .textContent(action + ' ' + items.length + ' item(s)?')
-    );
-  }
+  $scope.lock = vm.lock;
+  const confirmAdd = confirmDialogFactory('Add whitelisting for');
+  const confirmRemove = confirmDialogFactory('Remove whitelisting for');
+  const  addDisabled = () => $scope.lock.locked || vm.selection.length === 0 || _.all(vm.selection, (doc)=> doc.whitelisted);
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBRejectsCtrl.js]: [nit] The spacing of the anonymous functions on this line and the following line is inconsistent. Can they all the update to have a single space between '()' and '=>'?

File: public/static/app/common/MuteDataService.js (hunk 1 above)
Comment (jimoleary) [reply public/static/app/common/MuteDataService.js]: Actually this comment in this location was left over when some some code was removed,

I'll update the comments where it should make sense (couple with an omit).

File: public/static/app/perf-bb/PerfBBOutliersCtrl.js (hunk 2 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBOutliersCtrl.js]: I'm happy to use what ever format is the best once we know what that is. I just copied the format I found in place (which tbh looks remarkebly like sphinx).

File: public/static/app/perf-bb/PerfBBOutliersCtrl.js (hunk 3 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBOutliersCtrl.js]: I will update to have a proper comment that explains the intent. (ditto for sorting).

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js (hunk 4 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: :+1:

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js (hunk 5 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: doh, I thought I'd gotten them all consistent.

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js (hunk 6 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: done and moved them up in terms of scope.

File: public/static/app/perf-bb/PerfBBOutliersCtrl.test.js (hunk 7 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBOutliersCtrl.test.js]: :+1:

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js (hunk 8 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: Looks like there is just some bad white space between the 2 lines. There should be a single space between the round brackets and the arrow,

**Pull request merged**